llmfoundry/eval/metrics/nlp.py

```diff
@@ -0,0 +1,605 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""A collection of common torchmetrics for NLP tasks."""
+
+import logging
+import os
+import re
+import string
+import warnings
+from typing import Any, Dict, List
+
+import numpy as np
+import torch
+from composer.utils import dist
+from composer.utils.eval_client import (EvalClient, LambdaEvalClient,
+                                        LocalEvalClient,
+                                        MosaicMLLambdaEvalClient)
+from torch import Tensor
+from torch.nn import functional as F
+from torchmetrics import Metric
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    'InContextLearningMetric',
+    'InContextLearningLMAccuracy',
+    'InContextLearningMultipleChoiceAccuracy',
+    'InContextLearningGenerationExactMatchAccuracy',
+    'InContextLearningCodeEvalAccuracy',
+    'InContextLearningLMExpectedCalibrationError',
+    'InContextLearningMCExpectedCalibrationError',
+]
+
+
+class InContextLearningMetric(Metric):
+
+    def __init__(self, *args, **kwargs):  # pyright: ignore
+        super().__init__(*args, **kwargs)
+        self.needs_batch = True
+
```

Comment:
We don't have `nlp.py` updates that correspond to the eval logging. @bmosaicml do you want me to add those?

Reply:
yes please do!

Reply:
I don't really have the capacity but I'll see what I can do

Reply:
I can do it tonight if not

Reply:
you resolved this, but I don't think made the changes? unresolving for now

Reply:
Yeah I think it would be best to put that in a follow on PR since it depends on a new composer release. wdyt?